### ISSUE TITLE
[BUG] MIMI Encoder produces different outputs for batched vs single inputs due to missing padding mask support

### ISSUE THREAD

**Issue by harshaljanjani** [huggingface/transformers#43377]
### System Info

* `transformers` version: `5.0.0.dev0`
* Platform: `Linux-5.15.167.4-microsoft-standard-WSL2-x86_64-with-glibc2.39`
* Python version: `3.12.3`
* `huggingface_hub` version: `1.3.2`
* `safetensors` version: `0.7.0`
* `accelerate` version: `1.12.0`
* Accelerate config: `not installed`
* DeepSpeed version: `not installed`
* PyTorch version (accelerator?): `2.9.1+cu128 (CUDA)`
* GPU type: `NVIDIA GeForce RTX 4060 Laptop GPU`

### Who can help?

@Rocketknight1

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```python
import torch
import numpy as np
from transformers import AutoModel, AutoFeatureExtractor
# Ref: https://huggingface.co/docs/transformers/en/model_doc/mimi
# https://huggingface.co/kyutai/mimi
model_id = "kyutai/mimi"
feature_extractor = AutoFeatureExtractor.from_pretrained(model_id)
# MimiModel
model = AutoModel.from_pretrained(model_id)
model.eval()
sampling_rate = feature_extractor.sampling_rate
audio1 = np.sin(2 * np.pi * 440 * np.linspace(0, 1.0, int(sampling_rate * 1.0))).astype(np.float32)
audio2 = np.sin(2 * np.pi * 554 * np.linspace(0, 0.5, int(sampling_rate * 0.5))).astype(np.float32)

with torch.no_grad():
    inputs1 = feature_extractor(audio1, sampling_rate=sampling_rate, return_tensors="pt")
    outputs1_single = model.encode(inputs1["input_values"], padding_mask=inputs1.get("padding_mask"))
    inputs2 = feature_extractor(audio2, sampling_rate=sampling_rate, return_tensors="pt")
    outputs2_single = model.encode(inputs2["input_values"], padding_mask=inputs2.get("padding_mask"))
    inputs_batch = feature_extractor([audio1, audio2], sampling_rate=sampling_rate, return_tensors="pt", padding=True)
    outputs_batch = model.encode(inputs_batch["input_values"], padding_mask=inputs_batch["padding_mask"])

seq_len1 = outputs1_single[0].shape[2]
outputs1_from_batch = outputs_batch[0][0:1, :, :seq_len1]
diff1 = torch.abs(outputs1_single[0] - outputs1_from_batch).max().item()
print(f"Sample 1 - Max difference (single vs batch): {diff1:.6e}")
seq_len2 = outputs2_single[0].shape[2]
outputs2_from_batch = outputs_batch[0][1:2, :, :seq_len2]
diff2 = torch.abs(outputs2_single[0] - outputs2_from_batch).max().item()
print(f"Sample 2 - Max difference (single vs batch): {diff2:.6e}")
```

### Expected behavior

The outputs for a sample should be identical regardless of whether it is processed individually or as part of a padded batch -- when tested currently, `MimiModel._encode_frame` does not contain appropriate logic to handle `padding_mask`, giving incorrect latents when padding is present.

**Current Output:**

```
Single encoding - Sample 1 shape: torch.Size([1, 32, 13])
Single encoding - Sample 2 shape: torch.Size([1, 32, 7])
Batch encoding shape: torch.Size([2, 32, 13])
Sample 1 - Max difference (single vs batch): 0.000000e+00
Sample 2 - Max difference (single vs batch): 1.500000e+03
```

**Expected Output:**

```
Single encoding - Sample 1 shape: torch.Size([1, 32, 13])
Single encoding - Sample 2 shape: torch.Size([1, 32, 7])
Batch encoding shape: torch.Size([2, 32, 13])
Sample 1 - Max difference (single vs batch): 0.000000e+00
Sample 2 - Max difference (single vs batch): 0.000000e+00
```
